include/sdf/Param.hh

```diff
@@ -291,50 +300,62 @@ namespace sdf
   template<typename T>
   bool Param::Get(T &_value) const
   {
-    try
+    T *value = std::get_if<T>(&this->dataPtr->value);
+    if (value)
+      _value = *value;
+    else
     {
-      if (typeid(T) == typeid(bool) && this->dataPtr->typeName == "string")
+      std::string valueStr = this->GetAsString();
+      bool success = false;
+      ParamPrivate::ParamVariant pv;
+
+      if constexpr (std::is_same_v<T, bool>)
```

Comment:
While reviewing https://github.com/ignitionrobotics/sdformat/pull/589/ I realized once this is forward ported, we'll have to update all the `ValueFromStringImpl` calls with an extra argument. Do you mind refactoring this function so that the C++ type to string mapping happens in a separate function and then we can have just one call to `ValueFromStringImpl` here?

Reply:
f0f9759ec9d4572db9c5fd9e0dd861e6f679a7cf